lib/loader/loader.js

```diff
@@ -217,6 +217,15 @@ var loader, define, requireModule, require, requirejs;
   };
 
   define = function(name, deps, callback) {
+    var module = registry[name];
+
+    // If a module for this name has already been defined and is in any state
+    // other than `new` (meaning it has been or is currently being required),
+    // then we return early to avoid redefinition.
+    if (module && module.state !== 'new') {
```

Comment:
This still allows for "packages" (by this I mean the convention of the top level path of a module) to be split across files if a package lazily requires internal modules.  I see though, that `unsee` does not actually require modules to be undefined, it also doesn't unsee dependencies in a ref counted way, is it isn't the inverse of `require` either.

I'd prefer to keep it simple, but given how unsee works today, I think this is good for now.